File: 백준/Bronze/10808. 알파벳 개수/알파벳 개수.py
S = input()
alphabets = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
count = [0 for x in range(26)]

for alphabet in S: # O(n)
    count[alphabets.index(alphabet)] += 1 #alphabets.index(alphabet) : O(26) , count[인덱스]로 찾기 -> O(1)

for x in count:
    print(x,end=" ")

File: 백준/Bronze/10808. 알파벳 개수/알파벳 개수.java
import java.io.*;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        String S = br.readLine();
        char[] word = S.toCharArray();

        int[] count = new int[26];

        for (char alphabet : word){
            int i = alphabet - 'a';
            count[i]++;
        }
        
        for (int i : count){
            sb.append(i);
            sb.append(" ");
        }
        
        System.out.println(sb);

    }
}